src/sentry/tasks/groupowner.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import absolute_import, print_function
+
+from datetime import timedelta
+
+from django.utils import timezone
+
+from sentry.models.groupowner import GroupOwner, GroupOwnerType
+from sentry.tasks.base import instrumented_task
+from sentry.utils.committers import get_serialized_event_file_committers
+
+
+PREFERRED_GROUP_OWNERS = 3
+PREFERRED_GROUP_OWNER_AGE = timedelta(days=1)
+
+
+@instrumented_task(
+    name="sentry.tasks.process_suspect_commits", default_retry_delay=60 * 5, max_retries=5
+)
+def process_suspect_commits(event, **kwargs):
+    can_process = True
+
+    owners = GroupOwner.objects.filter(
+        group_id=event.group_id, project=event.project, organization=event.project.organization
+    )
+    owner_count = owners.count()
+    oldest_owner = None
+
+    if owner_count >= PREFERRED_GROUP_OWNERS:
+        # We have enough owners already - so see if any are old.
+        # If so, we can delete it and replace with a fresh one.
+        oldest_owner = (
+            owners.filter(date_added__lte=timezone.now() - PREFERRED_GROUP_OWNER_AGE)
+            .order_by("-date_added")
+            .first()
+        )
+        if not oldest_owner:
+            can_process = False
+
+    if can_process:
+        committers = get_serialized_event_file_committers(event.project, event)
```

Comment:
Ideally this would use the project fetched from the cache instead of a relationship read.